packages/zoe/src/z2spec/escrow-exo.js

```diff
@@ -0,0 +1,311 @@
+/**
+ * @file escrow exchange using ownable seat, resolver
+ */
+// @ts-check
+
+import { AmountMath } from '@agoric/ertp';
+import { deeplyFulfilledObject } from '@agoric/internal';
+import { prepareDurablePublishKit } from '@agoric/notifier';
+import { assert, bare, Fail, q } from '@endo/errors';
+import { E } from '@endo/far';
+import { M, objectMap } from '@endo/patterns';
+import {
+  addToAllocation,
+  subtractFromAllocation,
+} from '../contractFacet/allocationMath.js';
+import { isOfferSafe } from '../contractFacet/offerSafety.js';
+import { prepareOwnable } from '../contractSupport/prepare-ownable.js';
+
+/**
+ * @import {Zone} from '@agoric/base-zone'
+ */
+
+const { keys, entries, fromEntries } = Object;
+const { isEmpty } = AmountMath;
+
+const objectFilter = (obj, pred) =>
+  fromEntries(entries(obj).filter(([k, v]) => pred(v, k)));
+
+/**
+ * @param {Zone} zone
+ * @param {<T>() => PublishKit<T>} makeDurablePublishKit
+ */
+export const prepareSeatKit = (zone, makeDurablePublishKit) =>
+  zone.exoClassKit(
+    'Seat Kit',
+    undefined,
+    proposal => {
+      /** @type {PublishKit<unknown>} */
+      const exitKit = makeDurablePublishKit();
+      /** @type {PublishKit<PaymentPKeywordRecord>} */
+      const payoutsKit = makeDurablePublishKit();
+      return {
+        proposal,
+        exitKit,
+        payoutsKit,
+        exiting: false,
+        /** @type {Allocation} */
+        currentAllocation: proposal.give,
+      };
+    },
+    {
+      readOnly: {
+        hasExited() {
+          return this.state.exiting;
+        },
+        getProposal() {
+          return this.state.proposal;
+        },
+        getCurrentAllocation() {
+          return this.state.currentAllocation;
+        },
+        getExitSubscriber() {
+          return this.state.exitKit.subscriber;
+        },
```

Comment:
Even for an exit subscriber, we should wonder what happens to an old exit subscriber on ownership transfer. It provides only info, not authority, so it isn't too weird for a previous owner to observe a future transfer. But it is weird, and IMO a privacy expectation violation if it happens by default.